### PR TITLE
Pegasus: remove various subdomain sites

### PR DESCRIPTION
**Pull request by breville**
A variety of subdomains on http://code.org are removed, and now redirect to other sites, as shown in this table:

from | to
--- | ---
https://ar.code.org | http://program.ar/la-hora-del-codigo
https://br.code.org | http://programae.org.br
https://italia.code.org | https://programmailfuturo.it
https://ro.code.org | https://adfaber.org/HourofCode
https://sg.code.org | https://code.org
https://tr.code.org | http://robincode.org
https://uk.code.org | https://code.org
https://za.code.org | https://codeforchange.co.za

### CODE REVIEW

File: cookbooks/cdo-varnish/attributes/default.rb [hunk 1]
```diff
@@ -26,6 +26,14 @@
...
     'cseducationweek.com' => 'csedweek.org',
     'cseducationweek.org' => 'csedweek.org',
+    'ar.code.org'     => 'program.ar/la-hora-del-codigo/',
+    'br.code.org'     => 'programae.org.br/',
+    'italia.code.org' => 'programmailfuturo.it',
+    'ro.code.org'     => 'adfaber.org/HourofCode/',
+    'sg.code.org'     => 'code.org',
+    'tr.code.org'     => 'robincode.org',
+    'uk.code.org'     => 'code.org',
+    'za.code.org'     => 'codeforchange.co.za/',
```
Comment (breville) [inline cookbooks/cdo-varnish/attributes/default.rb]: @wjordan Is this the right place to do such redirects?

If so, what's the best way to test that this works?  

And is it safe to include the removal of these sites from pegasus in this PR as well?

Comment (wjordan) [reply cookbooks/cdo-varnish/attributes/default.rb]: I haven't touched or looked at any of this code since 2015 so I can't answer any of your questions confidently, but I'll give my best guesses.

- I think this is an OK place to do redirects
- Test manually after merging the attribute changes- this feature was never developed in a way that could be tested in an isolated environment. Note that a redirect for `ar.code.org` won't also provide a redirect for `staging-ar.code.org` here, so you'll need to include those if you want to test on staging, otherwise just test once it's pushed to production.

- Removing the subdomains from `CDO.partners` will remove them from the CloudFront distribution, which will cause those requests to fail unless you manually update the DNS entries first to send requests directly to the load balancer instead of through the CloudFront distribution. Each partner has the following manual DNS entries:

- `staging-za.code.org`: `ALIAS staging.code.org.`
- `test-za.code.org`: `ALIAS test.code.org.`
- `za.code.org`: `ALIAS code.org.`
- `localhost.za.code.org`: `ALIAS localhost.code.org.`

I'd suggest deleting all but the production one, which should have its alias updated to point to the `production-codeorg` load balancer.

Beyond that, I can't think of anything else that will definitely break from removing the partner sites in this PR, but I can't guarantee safety beyond whatever our tests currently cover.

**Comment by caleybrock** [comment]
CC @davidsbailey - great news for pegasus / content pipeline!

**Comment by davidsbailey** [comment]
screen shot 2018-06-11 at 9 45 31 am
excellent!!!

**Comment by breville** [comment]
@Erin007 That was a good call, and exactly what I've been doing in https://github.com/code-dot-org/code-dot-org/pull/22976.  Thanks.

**Comment by breville** [comment]
Folks, now that https://github.com/code-dot-org/code-dot-org/pull/22865 is working on production (see https://ar.code.org), this PR is ready for final review.

File: lib/test/cdo/test_url_converter.rb [hunk 2]
```diff
@@ -15,10 +15,8 @@ def test_no_ops_in_production_configuration
     assert_equal 'https://hourofcode.com', url_converter.replace_origin('https://hourofcode.com')
     assert_equal 'https://csedweek.org', url_converter.replace_origin('https://csedweek.org')
     assert_equal 'https://csedweek.org/about', url_converter.replace_origin('https://csedweek.org/about')
-    assert_equal 'https://uk.code.org', url_converter.replace_origin('https://uk.code.org')
-    assert_equal 'https://uk.code.org/about', url_converter.replace_origin('https://uk.code.org/about')
-    assert_equal 'https://ro.code.org', url_converter.replace_origin('https://ro.code.org')
-    assert_equal 'https://ro.code.org/about', url_converter.replace_origin('https://ro.code.org/about')
+    assert_equal 'https://advocacy.code.org', url_converter.replace_origin('https://advocacy.code.org')
```
Comment (caleybrock) [inline lib/test/cdo/test_url_converter.rb:18]: this change seems unrelated to this PR?

Comment (breville) [reply lib/test/cdo/test_url_converter.rb:18]: Given that we were losing tests on `uk.code.org` and `ro.code.org` I wanted to add a couple additional tests on a subdomain that remained, `advocacy.code.org`.
